tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -13615,6 +13629,447 @@ TEST_F(ExecutionTest, CoopVec_OuterProduct) {
+    memset(&ScissorRect, 0, sizeof(ScissorRect));
+    ScissorRect.right = static_cast<long>(RtDesc.Width);
+    ScissorRect.bottom = static_cast<long>(RtDesc.Height);
+    CommandList->SetGraphicsRootSignature(RootSignature);
+    CommandList->SetGraphicsRootDescriptorTable(0, ResHandle);
+    CommandList->SetGraphicsRootUnorderedAccessView(
+        1, AtomicCounterResource->GetGPUVirtualAddress());
+    CommandList->RSSetViewports(1, &Viewport);
+    CommandList->RSSetScissorRects(1, &ScissorRect);
+
+    // Indicate that the buffer will be used as a render target.
+    RecordTransitionBarrier(CommandList, RenderTarget,
+                            D3D12_RESOURCE_STATE_COPY_DEST,
+                            D3D12_RESOURCE_STATE_RENDER_TARGET);
+
+    CD3DX12_CPU_DESCRIPTOR_HANDLE RtvHandle(
+        RtvHeap->GetCPUDescriptorHandleForHeapStart(), 0, RtvDescSize);
+    CommandList->OMSetRenderTargets(1, &RtvHandle, FALSE, nullptr);
+
+    CommandList->ClearRenderTargetView(RtvHandle, ClearColor, 0, nullptr);
+    CommandList->IASetPrimitiveTopology(D3D_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
+    CommandList->DrawInstanced(3, 1, 0, 0);
+  }
+
+  RecordTransitionBarrier(CommandList, AccumulationVecUAVResource,
+                          D3D12_RESOURCE_STATE_UNORDERED_ACCESS,
+                          D3D12_RESOURCE_STATE_COPY_SOURCE);
+  CommandList->CopyResource(AccumulationVecUAVReadResource,
+                            AccumulationVecUAVResource);
+
+  CommandList->Close();
+  ExecuteCommandList(CommandQueue, CommandList);
+  WaitForSignal(CommandQueue, FO);
+
+  VERIFY_SUCCEEDED(CommandAllocator->Reset());
+  VERIFY_SUCCEEDED(CommandList->Reset(CommandAllocator, PipelineState));
+
+  {
+    MappedData MappedData(
+        AccumulationVecUAVReadResource,
+        static_cast<UINT>(AccumulationVector.getTotalBytes()));
+
+    void *ResultBuffer = MappedData.data();
+    bool Equal = true;
+
+    for (size_t i = 0; i < Config.VecSize; i++) {
+      float Result = 0.0f;
+      float Expected = 0.0f;
+
+      if (AccumulateProps.InputType == D3D12_LINEAR_ALGEBRA_DATATYPE_FLOAT32) {
+        Result = (reinterpret_cast<float *>(ResultBuffer))[i];
+        Expected =
+            (reinterpret_cast<float *>(ExpectedOutputVector.getBuffer()))[i];
+      } else if (AccumulateProps.InputType ==
+                 D3D12_LINEAR_ALGEBRA_DATATYPE_FLOAT16) {
+        Result = ConvertFloat16ToFloat32(
+            (reinterpret_cast<DirectX::PackedVector::HALF *>(ResultBuffer))[i]);
+        Expected = ConvertFloat16ToFloat32(
+            (reinterpret_cast<DirectX::PackedVector::HALF *>(
+                ExpectedOutputVector.getBuffer()))[i]);
+      }
+
+      if (isnan(Result) || isnan(Expected) ||
+          fabs(Result - Expected) > 0.00001) {
+        LogErrorFmt(L"Result mismatch at index %zu", i);
+        LogErrorFmt(L"Result: %f, Expected: %f", Result, Expected);
+        Equal = false;
+        break;
+      }
+    }
+    VERIFY_IS_TRUE(Equal);
```

Comment:
>VERIFY_IS_TRUE(Equal); [](http://example.com/codeflow?start=4&length=22)

You can get rid of Equal and this verify. The LogErrorFmt uses Log::Error, which will already mark the test case as failed.